api/apiclient.go

```diff
@@ -587,22 +587,20 @@ func newWebsocketDialer(cfg *websocket.Config, opts DialOpts) func(<-chan struct
 				return nil, parallel.ErrStopped
 			default:
 			}
-			logger.Infof("dialing %q", cfg.Location)
+			logger.Debugf("dialing %q", cfg.Location)
 			conn, err := opts.DialWebsocket(cfg)
 			if err == nil {
+				logger.Debugf("successfully dialed %q", cfg.Location)
 				return conn, nil
 			}
-			if isX509Error(err) {
+			if certErr := isX509Error(err); !a.HasNext() || certErr {
```

Comment:
isCertErr would be a clearer variable name to not make you think that this was an error object, but is a boolean.